xbmc/utils/AMLUtils.cpp

```diff
@@ -41,295 +41,200 @@
+  if (!SysfsUtils::HasRW("/sys/class/audiodsp/digital_raw"))
+  {
+    CLog::Log(LOGERROR, "AML: no rw on /sys/class/audiodsp/digital_raw");
+    return false;
+  }
 
-  if (has_hevc_10bit == -1)
+  if (!SysfsUtils::HasRW("/sys/class/amhdmitx/amhdmitx0/config"))
   {
-    CRegExp regexp;
-    regexp.RegComp("hevc:.*10bit");
-    std::string valstr;
-    if (SysfsUtils::GetString("/sys/class/amstream/vcodec_profile", valstr) != 0)
-      has_hevc_10bit = 0;
-    else
-      has_hevc_10bit = (regexp.RegFind(valstr) >= 0) ? 1 : 0;
+    CLog::Log(LOGERROR, "AML: no rw on /sys/class/amhdmitx/amhdmitx0/config");
+    return false;
   }
-  return (has_hevc_10bit == 1);
-}
 
-AML_SUPPORT_H264_4K2K aml_support_h264_4k2k()
-{
-  static AML_SUPPORT_H264_4K2K has_h264_4k2k = AML_SUPPORT_H264_4K2K_UNINIT;
+  if (!SysfsUtils::HasRW("/sys/class/vfm/map"))
+  {
+    CLog::Log(LOGERROR, "AML: no rw on /sys/class/vfm/map");
+    return false;
+  }
 
-  if (has_h264_4k2k == AML_SUPPORT_H264_4K2K_UNINIT)
+  if (!SysfsUtils::HasRW("/sys/class/tsync/enable"))
   {
-    std::string valstr;
-    if (SysfsUtils::GetString("/sys/class/amstream/vcodec_profile", valstr) != 0)
-      has_h264_4k2k = AML_NO_H264_4K2K;
-    else if (valstr.find("h264:4k") != std::string::npos)
-      has_h264_4k2k = AML_HAS_H264_4K2K_SAME_PROFILE;
-    else if (valstr.find("h264_4k2k:") != std::string::npos)
-      has_h264_4k2k = AML_HAS_H264_4K2K;
-    else
-      has_h264_4k2k = AML_NO_H264_4K2K;
+    CLog::Log(LOGERROR, "AML: no rw on /sys/class/tsync/enable");
+    return false;
   }
-  return has_h264_4k2k;
-}
 
-bool aml_support_vp9()
-{
-  static int has_vp9 = -1;
+  if (!SysfsUtils::HasRW("/sys/devices/system/cpu/cpu0/cpufreq/scaling_min_freq"))
+  {
+    CLog::Log(LOGERROR, "AML: no rw on /sys/devices/system/cpu/cpu0/cpufreq/scaling_min_freq");
+    return false;
+  }
 
-  if (has_vp9 == -1)
+  if (!SysfsUtils::HasRW("/sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq"))
   {
-    CRegExp regexp;
-    regexp.RegComp("vp9:.*compressed");
-    std::string valstr;
-    if (SysfsUtils::GetString("/sys/class/amstream/vcodec_profile", valstr) != 0)
-      has_vp9 = 0;
-    else
-      has_vp9 = (regexp.RegFind(valstr) >= 0) ? 1 : 0;
+    CLog::Log(LOGERROR, "AML: no rw on /sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq");
+    return false;
+  }
+
+  if (!SysfsUtils::HasRW("/sys/devices/system/cpu/cpu0/cpufreq/scaling_governor"))
+  {
+    CLog::Log(LOGERROR, "AML: no rw on /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor");
+    return false;
+  }
+
+  if (aml_has_frac_rate_policy() && !SysfsUtils::HasRW("/sys/class/amhdmitx/amhdmitx0/frac_rate_policy"))
+  {
+    CLog::Log(LOGERROR, "AML: no rw on /sys/class/amhdmitx/amhdmitx0/frac_rate_policy");
+    return false;
+  }
+
+  if (!SysfsUtils::HasRW("/sys/module/di/parameters/bypass_prog"))
+  {
+    CLog::Log(LOGERROR, "AML: no rw on /sys/module/di/parameters/bypass_prog");
+    return false;
```

Comment:
same here. also SysfsUtils::Set*() is never used on all these nodes, so perhaps they should be completely removed ?